densephrases/models/model_encoder.py

```diff
@@ -0,0 +1,275 @@
+import json
+import argparse
+import torch
+import os
+import random
+import numpy as np
+import requests
+import logging
+import math
+import copy
+import wandb
+import string
+
+from time import time
+from tqdm import tqdm
+
+from .encoder import DensePhrases
+
+class MaxSimEncoder(DensePhrases):
+    def __init__(self, *args, **kwargs):
+        super(MaxSimEncoder, self).__init__(*args, **kwargs)
+        self.query_encoder = copy.deepcopy(self.phrase_encoder)
+
+    def embed_phrase(self, input_ids, attention_mask, token_type_ids):
+        """ Get phrase embeddings (token-wise) """
+        
+        outputs_s = self.phrase_encoder(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+        )
+        
+        sequence_output_s = outputs_s[0]
+        sequence_output_e = outputs_s[0]
+        start = sequence_output_s[:,:,:]
+        end = sequence_output_e[:,:,:]
```

Comment:
did we figure out why these are exactly the same thing?

Reply:
For `embed_phrase`, it's only used when creating the phrase dump in `run_single`. `outputs_s[0]` is `batch_size x phrase_len x emb_dim`, and it doesn't appear to matter which one is start and end since the whole phrase embedding is stored anyways. Seems like the only reason a tuple is returned is to keep the API consistent with the `embed_query`?